api_app/ingestors_manager/ingestors/malware_bazaar.py

```diff
@@ -0,0 +1,138 @@
+import io
+import logging
+from datetime import datetime
+from typing import Any, Iterable
+from unittest.mock import patch
+
+import pyzipper
+import requests
+
+from api_app.ingestors_manager.classes import Ingestor
+from api_app.ingestors_manager.exceptions import IngestorRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections
+
+logger = logging.getLogger(__name__)
+
+
+class MalwareBazaar(Ingestor):
+    # API endpoint
+    url: str
+    # Download samples that are up to X hours old
+    hours: int
+    # Download samples from chosen signatures (aka malware families)
+    signatures: str
+
+    def run(self) -> Iterable[Any]:
```

Comment:
This function imho should be split in 3 parts:
- Where we are retrieving the results of each signatures
- Where we are parsing the results and adding stuff in the `hashes` variable
- Where we are downloading the file from the hashes
I would split this function in these 3 parts, to decrease its complexity